### DISCUSSION TITLE
Email Magic Link Authentication Not Working anymore

### DISCUSSION THREAD

**Discussion by fabiansiegel** [outline/outline#6996] [Self Hosting] [answered]
We have been using outline without problem for several months with both a keycloak integration and logins via the magic link support.
Not sure when, but about 2-6 weeks ago, magic link login has stopped being offered as a login method.

email server is setup correctly (and unchanged for months), emails are working correctly.
```
SMTP_HOST=relays.example.com
SMTP_PORT=25
SMTP_USERNAME=
SMTP_PASSWORD=
SMTP_FROM_EMAIL=operations@example.com
SMTP_REPLY_EMAIL=
SMTP_TLS_CIPHERS=
SMTP_SECURE=false
```

The setting "Allow members to sign-in using their email address" is checked correctly

![grafik](https://github.com/outline/outline/assets/4614435/ed8e9a66-5da5-40c0-a3f7-1b84616e4094)

I didn't see anything in the logs, but I can provide anything needed.

I'm not sure what triggered the problem, it might have been an update, but I only received delayed complaints.

Any ideas on what to do?

Thanks!

**Comment by tommoor** [comment, marked as answer]
There is a bug in `v0.76.0` that incorrectly assumed a valid SMTP configuration would include a username, there is a fix in `v0.76.2-0` prerelease so the easiest thing to do is update to that :)

**Comment by ovizii** [comment]
I'm so glad I found this. I thought I was going nuts, all of a sudden my only authentication method had disappeared. I just went ahead and migrated to OIDC only to find this thread. At least now I know it's not me who messed something up, it was a bug. :-)

**Comment by IamTaoChen** [comment]
It doesn't work for me. I used OIDC before and I want to change to email login now.
<img width="704" alt="image" src="https://github.com/outline/outline/assets/42793494/5ca09045-40f4-4f31-8f3d-93c2db384b0d">

**Reply by GabrielMoraez** [reply]
any news on this?

**Comment by RyanEwen** [comment]
I have the same issue as @IamTaoChen 

I have Discord and want to enable magic link as secondary but it just tries to use Discord after entering an email address.

If I disable Discord and try to use only the magic link, it breaks even more. Same errors as show in his post above.

**Reply by tommoor** [reply]
You would need to find the row in the `authentication_providers` table for discord login and change `enabled=FALSE`. At some point we will have UI for this

**Reply by RyanEwen** [reply]
Thanks @tommoor , that did it. 

I got the impression that a user should be able to login with any method that matches their email address, but I guess it defaults to the one they first (or last?) used, if enabled?

**Reply by tommoor** [reply]
If SSO is connected to a user account it forces SSO as a security measure
